### PR TITLE
Including fetch

### PR DESCRIPTION
**Pull request by thestrauss3**

### CODE REVIEW

File: app/controllers/api/fortunes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      "You will be awarded some great honor.",
+      "Tuesday After Lunch is the cosmic time of the week.",
+      "Blow it out your ear.",
+      "You definitely intend to start living sometime soon.",
+      "Don't tell any big lies today.  Small ones can be just as effective.",
+      "Don't worry, be happy.",
+      "Another good night not to sleep in a eucalyptus tree."
+    ]
+  end
+end
```
Comment (nwalberts) [inline app/controllers/api/fortunes_controller.rb]: You should not have this in your app?

File: app/controllers/restaurants_controller.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def show
   end
 
   def new
-    @restaurant = Restaurant.new
+    @restaurant = Restaurant.new()
```
[the remark this answers is not in the document]
Comment (nwalberts) [reply app/controllers/restaurants_controller.rb]: Please fix this!

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@ def index
 
   def new
     @burger = Burger.find(params[:burger_id])
-    @review = Review.new
+    @review = Review.new()
```
[the remark this answers is not in the document]
Comment (nwalberts) [reply app/controllers/reviews_controller.rb]: please fix this!

File: app/views/burgers/show.html.erb [hunk 4]
```diff
@@ -6,7 +6,10 @@
 <% if @burger.image_url %>
   <%= image_tag @burger.image_url, id: "burger_image"%>
 <% end %>
+
+
 <% unless @burger.reviews.empty? %>
+
```
Comment (nwalberts) [inline app/views/burgers/show.html.erb]: Please remove extra lines

File: app/views/static_pages/index.html.erb [hunk 5]
```diff
@@ -1,3 +1,10 @@
...
+
+<div id="app">
+  <h1>Home Page </h1>
+  <h1>Nine to Five Guys</h1>
+  <h3>Nine Burgers to Know</h3>
+</div>
+
+
+<div id="something">
+</div>
```
Comment (nwalberts) [inline app/views/static_pages/index.html.erb]: What is this div doing?

File: config/routes.rb [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
-  
+
   root 'static_pages#index'
   resources :burgers, only: [:index, :show] do
     resources :reviews, only: [:index, :new, :create]
   end
   resources :reviews, only: [:show]
+  post 'burgers' => 'burgers/new'
+  post 'reviews' => 'reviews/new'
+  post 'restaurants' => 'restaurants/new'
```
Comment (nwalberts) [inline config/routes.rb]: what are these routes doing?

File: db/seeds.rb [hunk 7]
```diff
@@ -8,9 +8,16 @@
...
+Restaurant.all.each do |r|
+  5.times do
+    FactoryGirl.create(:burger, restaurant: r)
+  end
+end
+Burger.all.each do |b|
+  3.times do
+    FactoryGirl.create(:review, burger: b)
+  end
+end
```
Comment (nwalberts) [inline db/seeds.rb]: Please rename "b" and "r" to more telling names

File: react/src/containers/App.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      <div>
+      <h1>show</h1>
+      <Router history={browserHistory}>
+        <Route path='/'>
+          <Route path='/burgers' component={BurgerIndex} />
+          <Route path="/burgers/:id" component={BurgerShow} />
+          <Route path="/reviews" component={ReviewIndex} />
+          <Route path='/restaurants' component={RestaurantIndex} />
+          <Route path='/restaurants/:id' component={RestaurantShow} />
+          <Route path='burgers/new' component={BurgerNew} />
```
Comment (nwalberts) [inline react/src/containers/App.js]: inconsistent routes (last route), please fix. 

You also are getting a bit ahead of yourself. This is a LARGE PR. In the future. E.g. make a PR with ONLY the index page, then ONLY the show page, etc

File: app/controllers/api/v1/burgers_controller.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+class Api::V1::BurgersController < ApiController
+  def show
+    @burger = Burger.all.sample
+    render json: @burger
+  end
```
Comment (enthusiastick) [inline app/controllers/api/v1/burgers_controller.rb:5]: So this action ignores the `:id` that is passed in through params, in favor of returning a random burger? Interesting.

File: app/controllers/api_controller.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class ApiController < ApplicationController
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  protect_from_forgery with: :null_session
```
Comment (enthusiastick) [inline app/controllers/api_controller.rb:4]: As a best practice, I prefer `protect_from_forgery unless: -> { request.format.json? }`. But this works too.

File: spec/features/burgers_features/burger_add_spec.rb [hunk 11]
```diff
@@ -1,17 +1,34 @@
...
+  end
+
+  scenario "User successfully adds a burger" do
+    visit new_restaurant_burger_path(restaurant)
 
     fill_in 'Name', with: "Double Garlic Steak Burger"
-    fill_in 'Price', with: "four dollars"
-    fill_in 'Description', with: "A seasame seed bun, swiss cheese, and two beef patties slathered in Garlic Butter"
+    fill_in 'Price', with: 4
+    fill_in 'Description', with: Faker::Internet.ip_v6_address
```
Comment (enthusiastick) [inline spec/features/burgers_features/burger_add_spec.rb:17]: This seems like a strange choice.

**Comment by enthusiastick** [comment]
This is a big refactor, touching several different parts of the code base. In the future, I would rather see PRs that focus on one feature at a time.

**Pull request merged**